src/api/spec/models/attrib_spec.rb

```diff
@@ -16,4 +16,38 @@
       expect(attribute.container).to be(package)
     end
   end
+
+  describe "validations" do
+    before do
+      subject.valid?
+    end
+
+    describe "#validate_value_count" do
+      let(:attrib_value) { build(:attrib_value, value: 'Random value 1') }
+      let(:attrib_allowed_value) { build(:attrib_allowed_value, value: 'Random value 2') }
+      let(:attrib_type) { create(:attrib_type_with_namespace, allowed_values: [attrib_allowed_value]) }
+
+      subject { build(:attrib, project: project, attrib_type: attrib_type, values: [attrib_value]) }
+
+      it { expect(subject.errors.full_messages).to match_array(["Values Value 'Random value 1' is not allowed. Please use one of: Random value 2"]) }
+    end
+
+    describe "#validate_issues" do
+      let(:issue_tracker) { create(:issue_tracker) }
+      let(:issue) { create(:issue, issue_tracker_id: issue_tracker.id) }
```

Comment:
the issue_tracker relation should be in the factory, then you can directly use `issue_tracker` instead of `issue_tracker_id`

And you could even create a factory inside `issue`,` issue_with_tracker`